html_css/v2/main-course/browser-compatibility.md

```diff
@@ -0,0 +1,70 @@
+
+Today, as the standards around the web evolve and change, and web developers begin implementing new features in their codebase, it is imperative for the browsers to provide support for those new features. If the user experience is impacted by the lack of support in the browsers, the users might find their way to the competitor.
+
+### When is It Safe to Use New Features
+As exciting as it is to implement new features, there is a risk to rushing. It would not be a positive experience for your users to find that, for example, your application used to work well in Firefox, but due to the changes in the codebase it is now unusable in Firefox but works well in Safari. Thankfully, there is a tool that can help you prevent this situation. 
+
+["Can I Use"](https://caniuse.com/) is a great resource to help you validate if new features are supported by the browsers. It provides statistics on which browsers and platforms are supporting new technologies, and even which versions of the browsers support specific features. 
+
+It is generally a good advice to implement new features when they are supported by the most common browsers. This way you're less likely to encounter an issue that a lot of users will face.
+
+### Mobile Browsers
+Traditionally, the Web was desktop computer first. The application was successful if it worked well on the desktop browsers. But as smartphones became more popular, with each year more and more users are using mobile devices as their main Web facing device. In some areas of the world, mobile users are a vast majority. 
+
+Mobile devices mostly consist of smartphones and tablets. The most popular mobile operating systems are Android and Apples iPhoneOS. 
+
+As you're developing your applications, you must also consider whether your application should be fully mobile compatible. There are a couple of specifics about mobile browsers that you need to keep in mind. 
+
+1. On iPhoneOS and iPadOS, Safari is technically the only supported browser. Yes, you can install Chrome or Firefox, and you can even set them as a default, but they are not full browsers. They are still using Safari rendering engine (WebKit). Therefore, in order for your web application to work for the Apple users, you have to assure support for WebKit and other technologies used in Safari. 
+2. Another consideration for mobile browsers is the magnitude of different screen sizes. It is virtually impossible to have every physical device available to test, and thankfully browsers provide a way to emulate other devices. The important piece to remember is that when, for example, you emulate an iPhone in Chrome, all that you're emulating is the screen size. Keep in mind that any specific considerations of the iPhoneOS will not be reproduceable. 
+
```

Comment:
I would add a note here that mobile browsers are not necessarily the same as their desktop countertops... so checking in desktop Safari does not automagically mean your page will work on an iphone.

Also, chrome has a button to simulate popular device sizes... but this is _not_ a device emulator.  i.e. your page working in desktop chrome, when you click the 'iphone 12' button, does not mean it will work the same on an actual iphone 12.